frontend/src/metabase/public/containers/PublicDashboard/PublicDashboard.unit.spec.tsx

```diff
@@ -70,6 +70,16 @@ describe("PublicDashboard", () => {
 
     expect(firstTab).toHaveAttribute("aria-selected", "true");
   });
+
+  it("should work with filter value starting from '0' (metabase#41483)", async () => {
+    // note: as all slugs this is ignored and we only use the id
+    await setup({
+      queryString: "?my-filter-value=01",
+    });
+
+    // should not throw runtime error and render dashboard content
+    expect(screen.getByText(DASHBOARD_TITLE)).toBeInTheDocument();
```

Comment:
I think the behaviour customers expect is that the filter will have value 1, should we make an expect on that too?
I think we'll need to so some more mocking to make the filter show up in the UI

Reply:
To get `01` in the URL, customer needs to enter `01` in the filter. If the initial result shows as `01` then I'd expect that he might expect it to still show `01` (like what it initially was before refreshing the page)
![image](https://github.com/metabase/metabase/assets/1937582/b4cd7aee-c44a-48f1-9af4-0cba59906b9e)